### ISSUE TITLE
Can not find Unity Capture Device in OBS

### ISSUE THREAD

**Issue by Emylion** [schellingb/UnityCapture#5]
I ran install.bat and it succeed.
But, I can't  find Unity Capture device in OBS with Viedio Capture Device.

**Comment by schellingb** [comment]
Hi there

Running "install.bat" should show 2 messages about successful DLL registrations.
Then in OBS there should be a "Unity Video Capture" device available.
Like this:
![unitycaptureinobs](https://user-images.githubusercontent.com/14200249/50540323-0916b980-0bd3-11e9-8e5e-93b5a1b989fd.png)

Can you post what version of Windows you are running (and if 32 or 64 bit) and what version of OBS (and if 32 or 64 bit). It should work in all kinds of variations of these but maybe there is some special case or bug.

**Comment by katai5plate** [comment]
Hi.
I suffered the same trouble as @Emylion .
It seems that "Install.bat" can be executed normally only on the C drive.

I ran "install.bat" on the D drive, but could not install it properly.

**Comment by schellingb** [comment]
Hi there
That's rather weird, but maybe related to your Windows user's permissions.

Can you try running the installation from D drive with admin privileges?
> If you have problems registering or unregistering, right click on the Install.bat and choose "Run as Administrator".

@Emylion mentioned that installation succeeded (assuming it showed the message box about it being installed successfully) but still it didn't show up in OBS. Which I'm still not sure why. Maybe it got resolved since then.

**Comment by katai5plate** [comment]
@schellingb Yes.

I started "Install.bat" from the D drive with administrator privileges and a success dialog was displayed, but it was not available in OBS.
I can't find "Unity Video Capture" in the "Device" list box in OBS.

But, when running with administrator rights on the C drive, “Unity VideoC apture” was added to the “Device” list box.

Details are as follows:

---

- Windows 10  Home x64
- Dell G7 7588
- Language setting: Japanese

|HDD Drive|Dialog|"Device" listbox in OBS|
|-|-|-|
|C|Success|Not found|
|C(admin)|Success|Exist|
|D|Success|Not found|
|D(admin)|Success|Not found|

- Dialogs (Japanese)
This displayed in all 4 cases.
![image](https://user-images.githubusercontent.com/22496143/65343226-778f2200-dc0f-11e9-99b2-fdfaf3abfcd8.png)
![image](https://user-images.githubusercontent.com/22496143/65343236-7d850300-dc0f-11e9-870a-d944bd3aaae3.png)

- OBS on failure (The installation success dialog is displayed)
![image](https://user-images.githubusercontent.com/22496143/65343717-a1951400-dc10-11e9-838e-55b9a8f618e2.png)

- OBS on success
![image](https://user-images.githubusercontent.com/22496143/65343818-eae56380-dc10-11e9-844e-26b02aaec915.png)

**Comment by schellingb** [comment]
Thank you for the details!

Maybe the location where the DLL is located is not fully accessible by the user so even if admin can register the DLL, when OBS is started as the user it is not able to use the DLL. I guess if OBS is started as Administrator as well it would work but that is not desirable.

Maybe in such cases it is best to move the DLL and Install.bat into a user directory such as `Desktop` or `My Documents` to be sure it can always be fully accessed by the user.

If that helps, a clarification should probably be added to the README file.

**Comment by FreyaHolmer** [comment]
I'm getting the same issue - "Unity Video Capture" is no longer in the Device list

the only thing I've done in between it working vs not working, is that I've used the OBS virtual camera once (the integrated one where you press "Start Virtual Camera" under "Start Recording"). I don't really see how that could matter but I've tried reinstalling as per the instructions in this thread but somehow it's still not showing up, running, everything as administrator ;-;

idk if I'm doing something wrong but it's been working perfectly fine up until now, with no changes to either unity or OBS, as far as I know

update: I pulled the latest version and now it works ;-; idk what happened there but I'm glad it works now!

**Comment by schellingb** [comment]
@FreyaHolmer Hmm, maybe OBS updated itself recently? #33 has all the details but in short some change somewhere (OBS? Windows? Unity??) started to cause dead-locks while using UnityCapture. OBS rightfully wanted to prevent this so they put the DLL of UnityCapture onto a block list. The issue got fixed on our side and it was easiest to just switch UnityCapture to use a new DLL from now on so it doesn't get blocked anymore by OBS (file name changed from "UnityCaptureFilter32bit.dll" and "UnityCaptureFilter64bit.dll" to "UnityCaptureFilter32.dll" and "UnityCaptureFilter64.dll").
Cheers, happy to have you as a user :-)

**Comment by FreyaHolmer** [comment]
yeah it was probably an OBS update that snuck in there somewhere! had no idea about the blocklist thing, makes sense it wouldn't work then!
